rpc/service/src/service.rs

```diff
@@ -634,8 +634,22 @@ impl RpcApi for RpcCoreService {
 
     /// Start sending notifications of some type to a listener.
     async fn start_notify(&self, id: ListenerId, scope: Scope) -> RpcResult<()> {
-        self.notifier.clone().start_notify(id, scope).await?;
-        Ok(())
+        match scope {
```

Comment:
with `matches` macro
```
if matches!(scope, Scope::UtxosChanged(ref utxos_changed_scope) if !self.config.unsafe_rpc && utxos_changed_scope.addresses.is_empty()) {...
```

Reply:
Nah, good enough as is.

Though it's always fun, let's stop for now exploring the 50 nuances of Rust syntax :)